DepotDownloader/ContentDownloader.cs

```diff
@@ -1295,5 +1281,49 @@ private static async Task DownloadSteam3Async(uint appId, List<DepotDownloadInfo
             }
 
         }
+
+        static void DumpManifestToTextFile( DepotDownloadInfo depot, ProtoManifest manifest )
+        {
+            StringBuilder manifestBuilder = new StringBuilder();
+            string txtManifest = Path.Combine( depot.installDir, string.Format( "manifest_{0}_{1}.txt", depot.id, depot.manifestId ) );
+            manifestBuilder.Append( string.Format( "Content Manifest for Depot {0}\n\n", depot.id ) );
+            manifestBuilder.Append( string.Format( "Manifest ID / date     : {0} / {1}\n", depot.manifestId, manifest.CreationTime ) );
+            
+            int numFiles = 0, numChunks = 0;
+            ulong uncompressedSize = 0, compressedSize = 0;
+
+            foreach ( var file in manifest.Files )
+            {
+                if ( file.Flags.HasFlag( EDepotFileFlag.Directory ) )
+                    continue;
+
+                numFiles++;
+                numChunks += file.Chunks.Count;
+
+                foreach ( var chunk in file.Chunks )
+                {
+                    uncompressedSize += chunk.UncompressedLength;
+                    compressedSize += chunk.CompressedLength;
+                }
+            }
+
+            manifestBuilder.Append( string.Format( "Total number of files  : {0}\n", numFiles ) );

```

Comment:
This will generate culture-specific output. Should this be invariant?

Reply:
How? I don't believe numbers are culture-specific by default.

Reply:
You're right, it ends up using a null formatter, and the docs state that it is simply a sequence of digits with an optional leading negative sign.